src/components/NanoContract/NanoContractTransactionHeader.component.js

```diff
@@ -0,0 +1,219 @@
+        <TouchableWithoutFeedback onPress={() => {}}>
+          <View style={styles.headerWrapper}>
+            <InfoContainer>
+              <TextValue title pb4>{getShortHash(tx.txId, 7)}</TextValue>
+              <TextLabel>{t`Transaction ID`}</TextLabel>
+            </InfoContainer>
+            {isShrank
+              && <HeaderShrank />}
+            {isExpanded()
+              && <HeaderExpanded tx={tx} />}
+          </View>
+        </TouchableWithoutFeedback>
+      </HathorHeader.Central>
+    </HathorHeader>
+  )
+};
+
+const HeaderShrank = () => (
+  <ArrowDownIcon />
+);
+
+/**
+ * It presents the expanded header of Nano Contract Transaction screen
+ * containing contextual information about the Nano Contract and the transaction.
+ *
+ * @param {Obejct} props
+ * @param {Obejct} props.tx Transaction data
+ */
+const HeaderExpanded = ({ tx }) => {
+  const baseExplorerUrl = useSelector((state) => state.networkSettings.explorerUrl);
+  const ncId = getShortHash(tx.ncId, 7);
+  const callerAddr = getShortContent(tx.caller, 7);
+  const hasFirstBlock = tx.firstBlock != null;
+
+  const navigatesToExplorer = () => {
+    const txUrl = `transaction/${tx.txId}`;
+    const explorerLink = combineURLs(baseExplorerUrl, txUrl);
+    Linking.openURL(explorerLink);
+  };
+
+  return (
+    <>
+      <View style={styles.wrapper}>
+        <InfoContainer>
+          <TransactionStatusLabel hasFirstBlock={hasFirstBlock} isVoided={tx.isVoided} />
+        </InfoContainer>
+        <InfoContainer>
+          <TextValue bold pb4>{ncId}</TextValue>
+          <TextLabel>{t`Nano Contract ID`}</TextLabel>
+        </InfoContainer>
+        <InfoContainer>
+          <TextValue bold pb4>{tx.ncMethod}</TextValue>
+          <TextLabel>{t`Blueprint Method`}</TextLabel>
+        </InfoContainer>
+        <InfoContainer>
+          <TextValue bold pb4>{getTimestampFormat(tx.timestamp)}</TextValue>
+          <TextLabel>{t`Date and Time`}</TextLabel>
+        </InfoContainer>
+        <InfoContainer lastElement>
+          <TextValue bold>{callerAddr}</TextValue>
+          {tx.isMine
+            && (
+            <View style={styles.headlineLabel}>
+              <Text style={styles.isMineLabel}>{t`From this wallet`}</Text>
+            </View>
+            )}
+          <TextLabel>{t`Caller`}</TextLabel>
+        </InfoContainer>
+        <TwoActionsWrapper>
+          <PrimaryTextButton title={t`See transaction details`} onPress={navigatesToExplorer} />
+        </TwoActionsWrapper>
+      </View>
+      {/* TODO: add <ArrowUpIcon /> when action list is ready to be showed */}
+    </>
+  )
+};
+
+/**
+ * Container for value and label pair components.
+ *
+ * @param {Object} props
+ * @param {Object} props.children
+ */
+const InfoContainer = ({ lastElement, children }) => (
+  <View style={[styles.infoContainer, lastElement && styles.lastElement]}>
+    {children}
+  </View>
+);
+
+/**
+ * It presents two button options inline.
```

Comment:
I didn't understand this docstring. This component has only one link, the "See transaction details"

Reply:
Ok, I've improve4d the docstring.